### PR TITLE
Virtual chunk containers

### PR DESCRIPTION
**Pull request by paraseba**

### CODE REVIEW

File: icechunk/src/virtual_chunks.rs [hunk 1]
```diff
@@ -279,14 +285,53 @@ impl ChunkFetcher for S3Fetcher {
...
+        match checksum {
+            Some(Checksum::LastModified(SecondsSinceEpoch(seconds))) => {
+                b = b.if_unmodified_since(aws_sdk_s3::primitives::DateTime::from_secs(
+                    *seconds as i64,
+                ))
+            }
+            Some(Checksum::ETag(etag)) => {
+                b = b.if_match(etag);
+            }
+            None => {}
```
Comment (mpiannucci) [inline icechunk/src/virtual_chunks.rs:297]: Glad the checksum is here and optional 👍

File: icechunk-python/src/lib.rs [hunk 2]
```diff
@@ -122,6 +120,31 @@ impl From<SnapshotMetadata> for PySnapshotMetadata {
     }
 }
 
+#[derive(FromPyObject)]
+enum ChecksumArgument {
```
Comment (paraseba) [inline icechunk-python/src/lib.rs:124]: @mpiannucci is this the right approach to take a python argument that can be any of these types?

**Review by mpiannucci — APPROVED** [review]
I really like this API. Its way better than what we have right now

File: icechunk-python/src/lib.rs (hunk 2 above)
Comment (mpiannucci) [reply icechunk-python/src/lib.rs:124]: Yes!

**Pull request merged**